sapphire/mysqlConnector/Connection.cpp

```diff
@@ -222,6 +234,47 @@ MYSQL* Mysql::Connection::getRawCon()
    return m_pRawCon;
 }
 
+void Mysql::Connection::pingLoop()
+{
+   // todo: this sleep is so dumb, initialise the loop somewhere outside the constructor
+   std::this_thread::sleep_for( std::chrono::seconds( 5 ) );
+   std::chrono::seconds timeout;
```

Comment:
Why not extract out below (237-263) into a method called getTimeout() and have it return std::chrono:seconds or btter yet a std::chrono::duration.  Have pingLoop() take the timeout and call it beforehand or move timeout a member variable.

Reply:
das tru